modules/library/main/src/main/java/org/geotools/feature/visitor/AverageVisitor.java

```diff
@@ -86,6 +87,20 @@ public void init(SimpleFeatureCollection collection) {
         return Arrays.asList(expr);
     }
 
+    @Override
+    public Optional<List<Class>> getResultType(List<Class> inputTypes) {
+        if (inputTypes == null || inputTypes.size() != 1)
+            throw new IllegalArgumentException(
+                    "Expecting a single type in input, not " + inputTypes);
+
+        Class type = inputTypes.get(0);
+        if (Number.class.isAssignableFrom(type)) {
+            return Optional.of(inputTypes);
```

Comment:
Did we want Double here? Like standard deviation below?

Reply:
Well spotted. I've fixed it and added tests, along with some other type related fixes. They are not used by the aggregation fix, but they better be right anyways, could be of use down the road.